src/entry.cpp

```diff
@@ -122,7 +122,11 @@ Entry Entry::getFinalEntry() const
   if (final_article.good()) {
     return final_article;
   }
-
+  // Prevent infinite loops.
+  if (article.isRedirect() &&
+      article.getIndex() == article.getRedirectIndex()) {
+    throw NoEntry();
+  }
```

Comment:
I think that this test should be in (or after) the `while`.

Here, it is only catching if the first redirection is redirecting to itself. But the loop could be later on, add a bit more complex (A->B->C->A).

The simpler would be to check that `loopCounter>0` after the `while`. If `loopCounter==0`, it means that we end the `while` because of too many steps. It is probably because of a infinite loop and we should throw `NoEntry`.
This way, we should any kind of infinite loop.

Reply:
I dropped the second commit I'd attempted and went back to pretty much the very first version of the change I had, namely just checking that final_article isn't a redirect, which should address the issue.